tasks/main.yml

```diff
@@ -0,0 +1,422 @@
+  rabbitmq_policy:
+    apply_to: queues
+    name: pubsub_sweeper
+    state: present
+    pattern: ".*"
+    tags:
+      # Unused queues are killed after 1000 * 60 * 60 * 31 milliseconds (~a month)
+      expires: 111600000
+      # Queues can use at most 1GB of storage
+      max-length-bytes: 1073741824
+    vhost: /odcs
+  tags:
+  - odcs
+
+- name: Create the odcs-admin user for the odcs vhost (prod)
+  run_once: true
+  delegate_to: "rabbitmq01.iad2.fedoraproject.org"
+  rabbitmq_user:
+    user: odcs-admin
+    password: "{{ (env == 'production')|ternary(rabbitmq_odcs_admin_password_production, rabbitmq_odcs_admin_password_staging) }}"
+    vhost: /odcs
+    configure_priv: .*
+    read_priv: .*
+    write_priv: .*
+  tags:
+  - odcs
+
+- name: Dump the admin password in a file for administrative operations
+  run_once: true
+  delegate_to: "rabbitmq01.iad2.fedoraproject.org"
+  copy:
+    dest: /root/.odcs-rabbitmqpass
+    content: "{{ (env == 'production')|ternary(rabbitmq_odcs_admin_password_production, rabbitmq_odcs_admin_password_staging) }}"
+    mode: 0600
+    owner: root
+    group: root
+  tags:
+  - odcs
+
+- name: Grant the admin user access to the odcs vhost
+  run_once: true
+  delegate_to: "rabbitmq01.iad2.fedoraproject.org"
+  rabbitmq_user:
+    user: admin
+    vhost: /odcs
+    configure_priv: .*
+    read_priv: .*
+    write_priv: .*
+    tags: administrator
+  tags:
+  - odcs
+
+- name: Grant the nagios-monitoring user access to the odcs vhost
+  run_once: true
+  delegate_to: "rabbitmq01.iad2.fedoraproject.org"
+  rabbitmq_user:
+    user: nagios-monitoring
+    vhost: /odcs
+    configure_priv: "^$"
+    read_priv: "^$"
+    write_priv: "^$"
+    tags: monitoring
+  tags:
+  - odcs
+
+# Create a user with:
+- name: Create a user for odcs access
+  run_once: true
+  delegate_to: "rabbitmq01.iad2.fedoraproject.org"
+  rabbitmq_user:
+    user: "odcs-private-queue{{ env_suffix }}"
+    vhost: /odcs
+    configure_priv: .*
+    write_priv: .*
+    read_priv: .*
+    state: present
+  tags:
+  - odcs
+#########
+
+
+
+- name: copy the odcs-celery-backend.service file.
+  template:
+    src: "etc/systemd/system/odcs-celery-backend.service.j2"
+    dest: /etc/systemd/system/odcs-celery-backend.service
+    owner: odcs-server
+    group: odcs-server
+    mode: 0640
+  notify:
+  - restart odcs-celery-backend
+  tags:
+  - odcs
+
+- name: copy the odcs-backend.conf tmpfiles.d file.
+  copy:
+    src: "etc/tmpfiles.d/odcs-backend.conf"
+    dest: /etc/tmpfiles.d/odcs-backend.conf
+    owner: odcs-server
+    group: odcs-server
+    mode: 0640
+  tags:
+  - odcs
+
+- name: Run systemd-tmpfiles --create
+  command: systemd-tmpfiles --create
+  args:
+    creates: /var/run/odcs-backend
+  notify:
+  - restart odcs-celery-backend
+  tags:
+  - odcs
+
+- name: copy the odcs logrotate.d config.
+  copy:
+    src: "etc/logrotate.d/odcs"
+    dest: /etc/logrotate.d/odcs
+  tags:
+  - odcs
+
+- name: create ODCS_TARGET_DIR
```

Comment:
Should we be mounting this from our NFS share?

Reply:
I am not sure, but to serve it from a frontend box, we need to mount this dir both on backend and frontend.